sentry-types/src/protocol/envelope.rs

```diff
@@ -37,9 +40,66 @@ pub enum EnvelopeError {
     InvalidItemPayload(#[source] serde_json::Error),
 }
 
-#[derive(Deserialize)]
-struct EnvelopeHeader {
+/// The supported [Sentry Envelope Headers](https://develop.sentry.dev/sdk/data-model/envelopes/#headers).
+#[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq)]
+pub struct EnvelopeHeaders {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     event_id: Option<Uuid>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    dsn: Option<Dsn>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    sdk: Option<ClientSdkInfo>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "ts_rfc3339_opt"
+    )]
+    sent_at: Option<SystemTime>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    trace: Option<DynamicSamplingContext>,
+}
+
+impl EnvelopeHeaders {
+    /// Creates empty Envelope headers.
+    pub fn new() -> EnvelopeHeaders {
+        Default::default()
+    }
+
+    /// Sets the Event ID.
+    #[must_use]
+    pub fn with_event_id(mut self, event_id: Option<Uuid>) -> Self {
+        self.event_id = event_id;
+        self
+    }
```

Comment:
[optional suggestion] If you expect that users never will need to clear the `event_id` field, you can potentially achieve a cleaner API by taking `Uuid` rather than `Option<Uuid>` (so you only have a `None` value when you skip calling this method). Same applies to the other `with_` functions here. I would recommend staying consistent for all of the `with_` functions, regardless of what you decide here

```suggestion
    pub fn with_event_id(mut self, event_id: Uuid) -> Self {
        self.event_id = Some(event_id);
        self
    }
```